### PR TITLE
Fix anchored scroll view behavior.

### LINKED ISSUE #202: Fetching more messages causes scroll view to jump
When new messages are fetched, scroll position should be maintained so that scrolling feels smooth.

This is broken right now (both on master and in the `cached` PR). The scroll position jumps to the top when new messages are fetched. Probably an issue with `AnchoredScrollView`

### PR DESCRIPTION
**Pull request by neerajwahi**
Scrolling no longer jumps when new messages are fetched.

As part of this PR, I've removed the custom version of RCTScrollView (AnchoredScrollView) and instead updated the project to point to a fork of React Native with the changes included directly into RCTScrollView (and exposed via an `anchorMode` property). The fork of React Native is based on the latest master, so I also upgraded RN from 0.39.2 to latest in this diff.

Fixes #202 and #182

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,11 +28,11 @@
     "moment": "^2.17.1",
     "react": "^15.4.1",
     "react-addons-shallow-compare": "^15.4.1",
-    "react-native": "^0.39.2",
+    "react-native": "neerajwahi/react-native#master",
```
Comment (neerajwahi) [inline package.json]: I would like to move this fork to `zulip/react-native` but I don't have fork access to the zulip org. @timabbott would you be able to help me with that?

Comment (timabbott) [inline package.json]: Created https://github.com/zulip/react-native and made you an admin!

Comment (neerajwahi) [inline package.json]: Awesome thanks!

**Comment by neerajwahi** [comment]
This actually doesn't quite fix it. I suspect there might be an issue because the message list is no longer immutable (so the underlying view doesn't remain intact when elements are added). Investigating

**Comment by neerajwahi** [comment]
This PR doesn't have any significant non-scrolling changes and has been open for a while so I'm going to merge it now. In particular, there are no changes to caching here.

**Pull request merged**